File: 2021/08/1.rs
fn main() {
    let input = include_str!("input.txt");
    let lines: Vec<(Vec<String>, Vec<String>)> = input
        .lines()
        .map(|line| {
            let parts: Vec<&str> = line.split(" | ").collect();
            let signals: Vec<String> = parts[0]
                .split(" ")
                .map(|signal| {
                    let mut chars: Vec<char> = signal.chars().collect();
                    chars.sort();
                    chars.iter().collect::<String>()
                })
                .collect();

            let outputs: Vec<String> = parts[1]
                .split(" ")
                .map(|output| {
                    let mut chars: Vec<char> = output.chars().collect();
                    chars.sort();
                    chars.iter().collect::<String>()
                })
                .collect();
            (signals, outputs)
        })
        .collect();

    let count: usize = lines.iter()
        .map(|(_, outputs)| {
            outputs.iter()
                .filter(|output| {
                    match output.len() {
                        2 | 3 | 4 | 7 => true,
                        _ => false,
                    }
                })
                .count()
        })
        .sum::<usize>();

    println!("{}", count);
}

File: 2022/04/1.rb
require 'set'

def covers(a1, a2, b1, b2)
  return (a1 <= b1 && a2 >= b2) || (b1 <= a1 && b2 >= a2)
end

puts File.readlines('input.txt').filter { |line|
  m = line.match /(\d+)-(\d+),(\d+)-(\d+)/
  covers(m[1].to_i, m[2].to_i, m[3].to_i, m[4].to_i)
}.count
